crates/hyperqueue/src/client/output/cli.rs

```diff
@@ -958,6 +960,10 @@ pub fn print_job_output(
 
     match task_selector {
         None | Some(Selector::All) => {
+            if !task_status.is_empty() {
```

Comment:
Probably we can just filter the tasks at the beginning of the function once to avoid doing it three times? To avoid the "task not found" warning for tasks that were found, but also were filtered, we can build a set of filtered tasks and only print the warning if the task is not inside that set.